x-pack/plugins/fleet/public/custom_logs_assets_extension.tsx

```diff
@@ -8,26 +8,25 @@
 import React from 'react';
 import { i18n } from '@kbn/i18n';
 
+import { getLogsLocatorsFromUrlService } from '@kbn/logs-shared-plugin/common';
+
 import { CustomAssetsAccordion } from './components/custom_assets_accordion';
 import type { CustomAssetsAccordionProps } from './components/custom_assets_accordion';
 import { useStartServices } from './hooks';
 import type { PackageAssetsComponent } from './types';
 
 export const CustomLogsAssetsExtension: PackageAssetsComponent = () => {
-  const { http, cloud } = useStartServices();
-  const isLogsUIAvailable = !cloud?.isServerlessEnabled;
-  // if logs ui is not available, link to discover
-  // TODO: move away from hardcoded link and use locators instead
-  const logStreamUrl = isLogsUIAvailable
-    ? http.basePath.prepend('/app/logs/stream')
-    : http.basePath.prepend('/app/discover');
+  const {
+    share: { url },
+  } = useStartServices();
+  const { logsLocator } = getLogsLocatorsFromUrlService(url);
```

Comment:
> [!NOTE]
> Is there any point in passing from this locator instead of directly using the logs explorer one? AFAIK, this locator does an extra step looking for an infra locator.
```ts
  const infraLogsLocator = this.locators.get<LogsLocatorParams>(INFRA_LOGS_LOCATOR_ID);
    if (infraLogsLocator) {
      return infraLogsLocator.getLocation(params);
    }
```

Is it necessary here and in the other usages?

Reply:
Nevermind, I see you removed the above code in https://github.com/elastic/kibana/pull/190835/files#diff-eae3e2549b72d5b69d2bf48d3746db91f323a4005e784d6682d93bb2ccc99babL24.

I think it still has relevance, should we also remove this `logsLocator` and directly rely on the logsExplorer one?

Reply:
I actually really liked this `getLogsLocatorsFromUrlService` and logs shared top level locator abstraction. Previously this "wrapper" was very useful in facilitating the infra vs logs explorer fork, and in the future I could see this becoming useful again as we potentially move away from logs explorer in favour of One Discover. It gives us a little layer to do translation if needed, and I do think that's valuable.